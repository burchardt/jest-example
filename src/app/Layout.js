import React from "react";
import { MyButton } from './MyButton';
import { MyHeader } from './MyHeader';

export default class Layout extends React.Component {
	render() {
	    return(
	        <div>
	            <MyHeader />
	            <MyButton />
	        </div>
	    );
	}
}