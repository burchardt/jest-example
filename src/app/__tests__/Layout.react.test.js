import React from 'react';
import renderer from 'react-test-renderer';

import Layout from '../Layout';

describe('<Layout />', () => {
    it('should render the whole layout', () => {
        const tree = renderer.create(
            <Layout />
        ).toJSON();
        // Generate and match snapshot
        expect(tree).toMatchSnapshot();
    });
});