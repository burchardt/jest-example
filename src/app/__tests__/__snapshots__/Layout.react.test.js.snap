// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the whole layout 1`] = `
<div>
  <h1>
    Header
  </h1>
  <div>
    <button
      className="btn"
      id="myButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightblue",
          "fontSize": "25px",
        }
      }
    >
      Open
    </button>
  </div>
</div>
`;
